drivers/sensor/iis2mdc/iis2mdc_spi.c

```diff
@@ -84,17 +78,18 @@ static int iis2mdc_spi_write(struct iis2mdc_data *data, uint8_t reg_addr,
 	};
 
 
-	if (len > 64) {
-		return -EIO;
-	}
-
-	if (spi_write(data->bus, spi_cfg, &tx)) {
+	if (spi_write(cfg->bus, spi_cfg, &tx)) {
 		return -EIO;
 	}
 
 	return 0;
 }
 
+stmdev_ctx_t iis2mdc_spi_ctx = {
+	.read_reg = (stmdev_read_ptr) iis2mdc_spi_read,
+	.write_reg = (stmdev_write_ptr) iis2mdc_spi_write,
+};
```

Comment:
Stray addition? I don't see this getting used anywhere.

Reply:
They are indeed used in the stmemsc module. This required by the stmemsc HAL i/f to map the hardware specific i2c/spi methods:

https://github.com/zephyrproject-rtos/hal_st/blob/master/sensor/stmemsc/iis2mdc_STdC/driver/iis2mdc_reg.h

```
typedef struct {
      /** Component mandatory fields **/
     stmdev_write_ptr  write_reg;                                                                                                                             
     stmdev_read_ptr   read_reg;
     /** Customizable optional pointer **/
     void *handle;
  } stmdev_ctx_t;
```

Reply:
How does the typedef in the header file show that this instance of the struct is getting used somewhere?

Reply:
> How does the typedef in the header file show that this instance of the struct is getting used somewhere?

Not sure I got the question properly. Every sensor in the STMEMSC HAL module is using a similar ctx structure that the HAL users (the zephyr ST sensor drivers) need to populate with the platform bus specific read/write routines. The module does not know whether the ctx structure is used somewhere or not. Do you see an issue in that?

Reply:
I think @MaureenHelm's point is that `iis2mdc_spi_ctx` is not referenced anywhere.  Unless the pointer to it is being constructed from some magic macro catenation?

Reply:
Ah, yes. Now I see. This declaration is not needed anymore. Thanks for catching it.
Sorry for confusion.

Reply:
This and all the `stmdev_ctx_t` instances in the existing drivers should be `static`: there's no need to access them by name from outside the driver implementation.

Reply:
I mean, THIS declaration:

```
stmdev_ctx_t iis2mdc_spi_ctx = {
	.read_reg = (stmdev_read_ptr) iis2mdc_spi_read,
	.write_reg = (stmdev_write_ptr) iis2mdc_spi_write,
};
```

Reply:
re-pushed. Thanks again!